libexec/basher-upgrade

```diff
@@ -35,3 +35,6 @@ fi
 
 cd "${BASHER_PACKAGES_PATH}/$package"
 git pull
+
+basher-_unlink "$package"
```

Comment:
This unlink must be done before we pull the new changes right? Otherwise we fail to remove links to binaries that were removed.

Reply:
I moved the pull, that makes sense. Afterwards I added an additional test to `basher-upgrade.bats`. 

I could use a second pair of eyes on it.. I kept getting conflicts when upgrading the repo to take in the commit with the deletion. The filemode was updated. 

Doing a reset seems like the sensible thing to me since we would want to rely on the origin.. but I'm thinking you should have seen this before. wdyt @juanibiapina ?

Reply:
Nice catch! But instead of using a reset, you instead need to set the proper permissions in the original bin. So add `chmod +x "bin/$exec"` to `create_exec`, probably around line 65.

Reply:
This is happening because link bins is making the file executable, even though the original might not be. This is probably incorrect, but a problem for a different PR.

Reply:
ahhhh there it is! I also agree changing the file mode should be on a different changeset if you want to move forward with that. For this pr, I updated the test helper so that I could clean up the upgrade script.